src/main.cpp

```diff
@@ -28,20 +28,36 @@ bool validParams( int ac, char **av ) {
     return ( valid_params );
 }
 
+
+Server* serv;
+
+static void sigHandler(int sigNum) {
+    serv->exit(sigNum);
+}
+
 bool IRCserver( int port, std::string const & pass ) {
     Server ircserver = Server( port, pass );
+    serv = &ircserver;
     try {
         ircserver.init();
     } catch ( std::exception const & e ) {
         std::cerr << "[" << e.what() << "]" << std::endl;
     }
     ircserver.run();
-    // ircserver.end();
 
     return ( true );
 }
 
 int main( int ac, char **av ) {
+    struct sigaction sigIntHandler;
+
+   sigIntHandler.sa_handler = sigHandler;
+   sigemptyset(&sigIntHandler.sa_mask);
+   sigIntHandler.sa_flags = 0;
+
+    if (sigaction(SIGINT, &sigIntHandler, NULL) == -1)
+        std::cout << "problem with handler" << std::endl; 
+        
```

Comment:
Maybe we can move this inisde `IRCserver ()` since we're using it to stop the server.